apps/analytics/app/api/event/route.ts

```diff
@@ -13,36 +14,27 @@ export const runtime = "edge";
 export async function POST(request: NextRequest) {
   const event = (await request.json()) as DispatchableAnalyticsEvent;
 
-  const timestamp = JSON.stringify(event.timestamp || new Date().toISOString());
+  let generalEvent: EventInsertType = {
+    type: event.type,
+    metadata: event.metadata,
+    timestamp: event.timestamp,
+    installationId: event.installationId,
+    isocode: event.geolocation?.countryCode,
+  };
 
-  // determine if this is an error and push it to the errors table
   if (event.type === "Error") {
-    const errorPayload = event.error;
-    try {
-      await sql`INSERT INTO Errors (message, details, stacktrace, timestamp, installationid, path) VALUES (${errorPayload.message}, ${errorPayload.details}, ${errorPayload.stacktrace}, ${timestamp}, ${event.installationId}, ${errorPayload.path});`;
-      return NextResponse.json(
-        { errorPayload },
-        { status: 200, headers: corsHeaders }
-      );
-    } catch (error) {
-      return NextResponse.json(
-        { error },
-        { status: 500, headers: corsHeaders }
-      );
-    }
+    generalEvent = {
+      ...generalEvent,
+      message: event.error.message,
+      name: event.error.name,
+      stack: event.error.stack,
+    };
```

Comment:
You are only inserting events if they are type Error!!

What about normal analytics events?!

The whole point of this refactor was to eliminate different codepaths for errors and other analytics events!!

Reply:
Please update or remove this review sir, this almost gave me a heart attack. 😄

Reply:
Haha! Okay, apologies. This turned out to be as a result of poor code style and not actually a bug.